cdap-client/src/main/java/co/cask/cdap/client/AuthorizationClient.java

```diff
@@ -198,10 +198,13 @@ private void executeExistingRolesRequest(Role role, HttpRequest request) throws
     }
   }
 
-  private HttpResponse executePrivilegeRequest(EntityId entityId, HttpRequest request) throws FeatureDisabledException,
-    UnauthenticatedException, IOException, NotFoundException, UnauthorizedException {
+  private HttpResponse executePrivilegeRequest(EntityId entityId, Principal principal, HttpRequest request) throws
+    FeatureDisabledException, UnauthenticatedException, IOException, NotFoundException, UnauthorizedException {
     HttpResponse httpResponse = doExecuteRequest(request, HttpURLConnection.HTTP_NOT_FOUND);
     if (HttpURLConnection.HTTP_NOT_FOUND == httpResponse.getResponseCode()) {
+      if (!listAllRoles().contains(new Role(principal.getName()))) {
```

Comment:
This does not seems right to me. 
AuthorizationClient is just a client it cannot be aware of implementation details of underlying ACL management.
Here you are checking for Principals existence with just roles and its working for Sentry since sentry is role based and its grant/revoke is done only on roles. Sentry 1.7 (or the coming future release) or other auth engines like ranger support granting/revoking access to user/group. This approach will fail in that case. 

The underlying auth provider should check for principal existence not this auth client.